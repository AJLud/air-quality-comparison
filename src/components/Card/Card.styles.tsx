import styled from 'styled-components';
import { motion } from 'framer-motion';

export const CardContainer = styled.div`
  height: 240px;
  width: 600px;
  background-color: white;
  border-radius: 10px;
  margin: 40px 40px;
  display: flex;
  justify-content: space-between;
  @media screen and (max-width: 630px) {
    width: 90%;
    height: 200px;
    margin: 20px 0;
    padding: 15px 0;
  }
`;

export const CloseIcon = styled(motion.span)`
  position: relative;
  color: black;
  font-size: 45px;
  margin-left: 10px;
  margin-right: 10px;
  margin-top: 6px;
  right: 0px;
  top: 0px;

  &:hover {
    color: #b8b8b8;
  }
  @media screen and (max-width: 630px) {
    font-size: 35px;
  }
`;

export const CardInfo = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  margin-left: 40px;
  @media screen and (max-width: 630px) {
    margin-left: 20px;
  }
`;
export const LastUpdated = styled.h4`
  margin: 5px;
  color: black;
  font-size: 1.2rem;
  font-weight: 600;
  @media screen and (max-width: 630px) {
    font-size: 0.9rem;
  }
`;
export const Location = styled.h3`
  margin: 5px;
  color: rgba(118, 57, 176, 1);
  font-size: 1.9rem;
  font-weight: 700;
  @media screen and (max-width: 630px) {
    font-size: 1.3rem;
  }
`;
export const City = styled.h4`
  margin: 5px;
  color: black;
  font-size: 1.5rem;
  font-weight: 400;
  @media screen and (max-width: 630px) {
    font-size: 1.1rem;
  }
`;

export const Values = styled.h4`
  margin: 5px;
  color: black;
  font-size: 1.5rem;
  font-weight: 600;
  @media screen and (max-width: 630px) {
    font-size: 1.1rem;
  }
`;
