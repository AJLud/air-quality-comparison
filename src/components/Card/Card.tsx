import React from 'react';
import styled from 'styled-components';
import { IoClose } from 'react-icons/io5';
import { motion } from 'framer-motion';

const CardContainer = styled.div`
  height: 240px;
  width: 600px;
  background-color: white;
  border-radius: 10px;
  margin: 40px;
  display: flex;
  justify-content: space-between;
`;

const CloseIcon = styled(motion.span)`
  position: relative;
  color: black;
  font-size: 45px;
  margin-left: 10px;
  margin-right: 10px;
  margin-top: 6px;
  right: 0px;
  top: 0px;

  &:hover {
    color: #b8b8b8;
  }
`;

const CardInfo = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  margin-left: 40px;
`;
const LastUpdated = styled.h4`
  margin: 5px;
  color: black;
  font-size: 1.2rem;
  font-weight: 600;
`;
const City = styled.h3`
  margin: 5px;
  color: rgba(118, 57, 176, 1);
  font-size: 1.9rem;
  font-weight: 700;
`;
const Location = styled.h4`
  margin: 5px;
  color: black;
  font-size: 1.5rem;
  font-weight: 400;
`;

const Values = styled.h4`
  margin: 5px;
  color: black;
  font-size: 1.5rem;
  font-weight: 600;
`;

const Card: React.FunctionComponent = () => (
  <CardContainer>
    <CardInfo>
      <LastUpdated>UPDATED AN HOUR AGO</LastUpdated>
      <City>Manchester Piccadilly</City>
      <Location>in Manchester, United Kingdom</Location>
      <Values>Values: PM25: 9, SO2: 32, O3: 8, NO2: 43</Values>
    </CardInfo>
    <CloseIcon>
      <IoClose />
    </CloseIcon>
  </CardContainer>
);

export default Card;
