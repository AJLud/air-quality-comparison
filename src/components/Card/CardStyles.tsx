import styled from 'styled-components';
import { motion } from 'framer-motion';

export const CardContainer = styled.div`
  height: 240px;
  width: 600px;
  background-color: white;
  border-radius: 10px;
  margin: 40px;
  display: flex;
  justify-content: space-between;
`;

export const CloseIcon = styled(motion.span)`
  position: relative;
  color: black;
  font-size: 45px;
  margin-left: 10px;
  margin-right: 10px;
  margin-top: 6px;
  right: 0px;
  top: 0px;

  &:hover {
    color: #b8b8b8;
  }
`;

export const CardInfo = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  margin-left: 40px;
`;
export const LastUpdated = styled.h4`
  margin: 5px;
  color: black;
  font-size: 1.2rem;
  font-weight: 600;
`;
export const City = styled.h3`
  margin: 5px;
  color: rgba(118, 57, 176, 1);
  font-size: 1.9rem;
  font-weight: 700;
`;
export const Location = styled.h4`
  margin: 5px;
  color: black;
  font-size: 1.5rem;
  font-weight: 400;
`;

export const Values = styled.h4`
  margin: 5px;
  color: black;
  font-size: 1.5rem;
  font-weight: 600;
`;
