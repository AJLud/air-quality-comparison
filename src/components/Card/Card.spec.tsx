/* eslint-disable no-undef */
import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import Card from './Card';

describe('SearchBar', () => {
  const mockSetState = jest.fn();
  const locationSelection: LocationResponse[] = [];
  const locationData = {
    location: 'Salford Eccles',
    city: 'Manchester',
    country: 'GB',
    coordinates: {
      latitude: 53.48481,
      longitude: -2.334139,
    },
    measurements: [
      {
        parameter: 'pm25',
        value: 24,
        lastUpdated: '2022-04-13T04:00:00+00:00',
        unit: 'µg/m³',
      },
      {
        parameter: 'no2',
        value: 18,
        lastUpdated: '2022-04-13T04:00:00+00:00',
        unit: 'µg/m³',
      },
      {
        parameter: 'pm10',
        value: 32,
        lastUpdated: '2022-04-13T04:00:00+00:00',
        unit: 'µg/m³',
      },
    ],
  };
  it('matches snapshot', () => {
    const { getByTestId } = render(
      <Card
        setLocationSelection={mockSetState}
        locationSelection={locationSelection}
        locationData={locationData}
      />,
    );
    expect(getByTestId('Card')).toMatchSnapshot();
  });
});
