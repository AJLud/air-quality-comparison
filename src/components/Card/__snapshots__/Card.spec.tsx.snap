// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar matches snapshot 1`] = `
<div
  class="sc-bczRLJ iqMnOt"
  data-testid="Card"
>
  <div
    class="sc-dkzDqf hkfOnF"
  >
    <h4
      class="sc-hKMtZM oUMZu"
    >
      12 HOURS AGO
    </h4>
    <h3
      class="sc-eCYdqJ kZfHnh"
    >
      Salford Eccles
    </h3>
    <h4
      class="sc-jSMfEi cetBCV"
    >
      in Manchester, United Kingdom
    </h4>
    <h4
      class="sc-gKXOVf fWIEzK"
    >
      Values: PM25: 24 NO2: 18 PM10: 32
    </h4>
  </div>
  <span
    class="sc-gsnTZi dOilkS"
  >
    <svg
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 512 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M289.94 256l95-95A24 24 0 00351 127l-95 95-95-95a24 24 0 00-34 34l95 95-95 95a24 24 0 1034 34l95-95 95 95a24 24 0 0034-34z"
      />
    </svg>
  </span>
</div>
`;
